src/tests/system/tests/test_passkey.py

```diff
@@ -0,0 +1,480 @@
+
+    client.sssd.pam['pam_passkey_auth'] = 'true'
+    client.sssd.start()
+
+    for record_file in [f'{tc}.ioctl', f'{tc}.script', f'{tc}.device']:
+        src_file = f"{os.getcwd()}/tests/data/passkey/{tc}/{record_file}"
+        client.tools.copy_files(src_file, f"/tmp/{record_file}")
+
+    output = client.auth.passkey.su(tc, username, 67890)
+    assert output.stdout_lines[-1] == 'su: Authentication failure'
+
+
+@pytest.mark.tier(1)
+@pytest.mark.topology(KnownTopology.IPA)
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_passkey__su_check_fail_wrong_passkey_string(client: Client, provider: GenericProvider):
+    """
+    :title: Check authentication deny when wrong passkey mapping is used while adding the user in IPA and LDAP server
+    :setup:
+        1. Add a IPA and LDAP user with passkey_mapping.
+        2. Setup IDM client with FIDO and umockdev setup
+        3. Create the recording files when authentications is failed due to wrong passkey mapping was added
+        4. Add 'pam_passkey_auth = True' under pam section of sssd.conf file.
+    :steps:
+        1. Check the umockdev-run command for authentication of the user.
+        2. Check umockdev-run that the output contains the authentication failure message
+    :expectedresults:
+        1. Command returns 0.
+        2. Output contains Authentication failure
+    :customerscenario: False
+    """
+    tc = passkey_str = username = ""
+    if isinstance(provider, IPA):
+        tc = "su_wrong_passkey_str_test5"
+        username = 'user1@ipa.test'
+        passkey_str = "passkey:oducA9WSTrzBHX2gUKylRNl2PD2XCb4a7V0XJOtahqIX7wGcAugflvrVjbWG2JPTsLlVf+j/dmi" \
+                      "a7SNIVhK5AA==,MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEGEa7EktmUw4AOR6Y6r1W2zxXptQh3YaDN" \
+                      "dvQEifZ3NpgRosVv+GS85uR3h6Ed1E7FtgfugwsZYeR8+9+GM6h8g=="
+
+    elif isinstance(provider, LDAP):
+        tc = "su_ldap_wrong_passkey_str_test8"
+        username = 'user1@test'
+        passkey_str = "passkey:aEgemlnC6a/WOoEZ8qU1YMwsTW9+uwmMsJnrgOXwTID0qIBHirzHp6d+e1d3WBhcSf7t9Ji8fl3AdSPtlb" \
+                      "dN5Q==,MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAENwDQHwyZmnYaUEp0UNqqnw0tGOGnqOMBGdds6O3+JKbmmJGT" \
+                      "n0vo7sKNNcDWDsFhJFU/RLWXmHXglxSo+yw9iQ=="
+
+    user = provider.user('user1').add()
+    user.passkey_add(passkey_str)
+
+    client.sssd.pam['pam_passkey_auth'] = 'true'
+    client.sssd.start()
+
+    for record_file in [f'{tc}.ioctl', f'{tc}.script', f'{tc}.device']:
+        src_file = f"{os.getcwd()}/tests/data/passkey/{tc}/{record_file}"
+        client.tools.copy_files(src_file, f"/tmp/{record_file}")
+
+    output = client.auth.passkey.su(tc, username, 123456)
+    assert output.stdout_lines[-1] == 'su: Authentication failure'
+
+
+@pytest.mark.tier(1)
+@pytest.mark.topology(KnownTopology.IPA)
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_passkey__su_check_when_srv_offline(client: Client, provider: GenericProvider):
```

Comment:
rename test_passkey__su_check_when_srv_not_resolvable (or something similar) to indicate it's a hostname/DNS resolution issue being tested.

Reply:
Kept the note of this.